--- lms/modules/invoicenew.php	2015-08-04 14:56:32.000000000 +0200
+++ htdocs/modules/invoicenew.php	2015-08-04 16:38:04.000000000 +0200
@@ -300,7 +300,12 @@ switch($action)
 			break;
 
 		$DB->BeginTrans();
-		$DB->LockTables(array('documents', 'cash', 'invoicecontents', 'numberplans', 'divisions'));
+		//Added for STCK by Sarenka MAXCON
+		if (ConfigHelper::getConfig('phpui.stock'))
+			$DB->LockTables(array('documents', 'cash', 'invoicecontents', 'numberplans', 'divisions','stck_stockassigments','stck_stock'));
+		else
+			 $DB->LockTables(array('documents', 'cash', 'invoicecontents', 'numberplans', 'divisions'));
+		//END STCK
 
 		if(!$invoice['number'])
 			$invoice['number'] = $LMS->GetNewDocumentNumber(DOC_INVOICE, $invoice['numberplanid'], $invoice['cdate']);
@@ -319,6 +324,17 @@ switch($action)
 		$invoice['type'] = DOC_INVOICE;
 		$iid = $LMS->AddInvoice(array('customer' => $customer, 'contents' => $contents, 'invoice' => $invoice));
 
+		//Added for STCK by Sarenka MAXCON
+		if (ConfigHelper::getConfig('phpui.stock')) {
+			foreach($contents as $ct) {
+				if ($ct['stckproductid']) {
+					$LMSST->StockSell($iid, $ct['stckproductid'], $ct['valuebrutto'], $invoice['cdate']);
+					}
+			}
+		}
+		//END STCK
+
+
 		// usuwamy wczesniejsze zobowiazania bez faktury
 		foreach ($contents as $item)
 			if (!empty($item['cashid']))
